include/glow/Graph/Graph.h

```diff
@@ -224,16 +230,19 @@ class Function final : public Named {
   Module *parent_;
 
   /// The log context associated with this function.
-  LogContext logCtx_;
+  std::shared_ptr<LogContext> logCtx_;
 
 public:
   Function(Module *parent, llvm::StringRef Name = {})
-      : Named(Name), parent_(parent) {}
+      : Named(Name), parent_(parent) {
+    logCtx_ = std::make_shared<LogContext>();
+    logCtx_.get()->setParent(this);
```

Comment:
you dont need the call to get() here. You can just use the -> on a shared_ptr.

Similar comments on a lot of cals below.